NAudio/Midi/MetaEvent.cs

```diff
@@ -44,6 +44,11 @@ public MetaEvent(MetaEventType metaEventType, int metaDataLength, long absoluteT
         }
 
         /// <summary>
+        /// Creates a deep clone of this MIDI event.
+        /// </summary>
+        public override MidiEvent Clone() => new MetaEvent(metaEvent, metaDataLength, AbsoluteTime) { data = (byte[])data?.Clone() };

```

Comment:
For example, this deep clone needs to be implemented manually in case `data` is used. However, `data` is not used by any subclass and it's not accessible either, so every subclass will have to override `Clone()` in order to duplicate the original implementation in `MidiEvent.Clone()`.
A solution here might be moving the `data` field to the `UnknownMetaEvent`, since it's null for all other meta events. That way `Clone()` wouldn't have to be overridden in all other `MetaEvent` subclasses.